12-offer-encoding.md

```diff
@@ -551,144 +585,100 @@ using the `onion_message` `invoice` field.
 ## Requirements
 
 A writer of an invoice:
-  - MUST set `created_at` to the number of seconds since Midnight 1
+  - if creating an `invoice` for an offer with `offer_send_invoice`:
+     - MUST copy all fields from the offer (including unknown fields).
+  - otherwise (responding to an `invoice_request`):
+     - MUST copy all non-signature fields from the invoice_request (including unknown fields).
+  - MUST set `invoice_created_at` to the number of seconds since Midnight 1
     January 1970, UTC when the offer was created.
-  - MUST set `payment_hash` to the SHA2 256-bit hash of the
+  - MUST set `invoice_payment_hash` to the SHA2 256-bit hash of the
     `payment_preimage` that will be given in return for payment.
-  - MUST set (or not set) `send_invoice` the same as the offer.
-  - MUST set `offer_id` to the id of the offer.
   - MUST specify exactly one signature TLV element: `signature`.
-    - MUST set `sig` to the signature using `node_id` as described in [Signature Calculation](#signature-calculation).
-  - if the chain for the invoice is not bitcoin:
-    - MUST specify `chain` the invoice is valid for.
-  - otherwise:
-    - the bitcoin chain is implied as the first and only entry.
-  - if it has bolt11 features:
-    - MUST set `features` to the bitmap of features.
-  - if the expiry for accepting payment is not 7200 seconds after `created_at`:
-    - MUST set `relative_expiry` `seconds_from_creation` to the number of
-      seconds after `created_at` that payment of this invoice should not be attempted.
-  - if the `min_final_cltv_expiry` for the last HTLC in the route is not 18:
-    - MUST set `min_final_cltv_expiry`.
+    - MUST set `sig` to the signature using `offer_node_id` as described in [Signature Calculation](#signature-calculation).
```

Comment:
I think this can only apply if we're responding to an `invoice_request`. If we're sending to a `send_invoice` offer, we don't have the `offer_node_id` privkey to sign with?

Reply:
Yes, this is fixed in the https://github.com/rustyrussell/lightning-rfc/pull/11